### PR TITLE
Remove tippex dependency and skip unnecessary sourcemap generation

### LINKED ISSUE #41: Plugin failures due to tippex dependency
Hi! I'm afraid the recently introduced dependency on `tippex` (https://github.com/surma/rollup-plugin-off-main-thread/pull/32) has broken our app's Rollup build in a few ways:
* tippex throws an error when running on some of our third party dependencies - for example [this line of code](https://github.com/visionmedia/page.js/blob/master/page.mjs#L800) in page.js:
```
[!] (plugin off-main-thread) Error: Unexpected character (820:3). If this is valid JavaScript, it's probably a bug in tippex. Please raise an issue at https://github.com/Rich-Harris/tippex/issues – thanks!
  };
    ^
../../../common/temp/node_modules/.pnpm/page@1.11.6/node_modules/page/page.mjs
Error: Unexpected character (820:3). If this is valid JavaScript, it's probably a bug in tippex. Please raise an issue at https://github.com/Rich-Harris/tippex/issues – thanks!
  };
    ^
    at find (<truncated>/tippex@3.0.0/node_modules/tippex/src/index.js:318:10)
    at Object.match (<truncated>/tippex@3.0.0/node_modules/tippex/src/index.js:366:16)
    at Object.transform (<truncated>/@surma/rollup-plugin-off-main-thread@2.0.0/node_modules/@surma/rollup-plugin-off-main-thread/index.js:114:14)
    at Object.timedPlugin.<computed> (<truncated>/rollup@2.36.1/node_modules/rollup/dist/shared/rollup.js:9695:45)
    at <truncated>/rollup@2.36.1/node_modules/rollup/dist/shared/rollup.js:18781:25
```
* A similar failure occurs when tippex runs on [this code](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L240) in URI.js. 
* Even after I manually patch the plugin's `transform` function to ignore the above files, or comment out the problematic code in those files, tippex still fails in the `renderChunk` function, in/around some LitElement copyright headers:
```
Error: Unexpected character (537:5). If this is valid JavaScript, it's probably a bug in tippex. Please raise an issue at https://github.com/Rich-Harris/tippex/issues – thanks!
    }    /**
      ^
    at find (<truncated>/tippex@3.0.0/node_modules/tippex/src/index.js:318:10)
    at Object.match (<truncated>/tippex@3.0.0/node_modules/tippex/src/index.js:366:16)
    at Object.renderChunk (<truncated>/@surma/rollup-plugin-off-main-thread@2.1.0/node_modules/@surma/rollup-plugin-off-main-thread/index.js:260:14)
    at <truncated>/rollup@2.36.1/node_modules/rollup/dist/shared/rollup.js:18781:25
```

In all cases it seems to be failing on valid JS code.

### Options to fix this
We could try to fix these apparent issues in tippex. However, given the number of issues already surfacing with it, I'm concerned that similar problems are liable to surface as our codebase grows. It's also not clear to me that tippex is actively maintained, as it has not been modified for a few years.

**For now, I wondered if you would be open to a PR that reverts to simple regex matching, instead of tippex?** Long term, if a more robust approach is desired, the plugin could switch over to AST, which seems common in other rollup plugins we use. It would be more expensive, but costs could be mitigated by allowing plugin users to specify `include` / `exclude` filters.

CC @surma @RReverser

### PR DESCRIPTION
**Pull request by davidclark87**
Fixes https://github.com/surma/rollup-plugin-off-main-thread/issues/41

### CODE REVIEW

**Comment by davidclark87** [comment]
@surma any chance you could take a look at this? 

Note the change should appear simpler if you ignore whitespace. 

I just added a small tweak to renderChunk to skip sourcemap generation if the code wasn't actually modified. This cuts my project's build time from around 45s to 30s.

File: index.js [hunk 1]
```diff
@@ -98,89 +97,81 @@ module.exports = function(opts = {}) {
...
-            workerFile = workerFile.slice(1, -1);
-
-            if (!/^\.{1,2}\//.test(workerFile)) {
+                matchIndex
+              );
+              longWarningAlreadyShown = true;
+            } else {
               this.warn(
-                `Paths passed to the Worker constructor must be relative to the current file, i.e. start with ./ or ../ (just like dynamic import!). Ignoring "${workerFile}".`,
+                `rollup-plugin-off-main-thread: Treating \`${fullMatchWithOpts}\` as \`${fullReplacement}\``,
```
Comment (RReverser) [inline index.js:137]: Why was this warning message changed? It provided more info than the new one.

Comment (RReverser) [reply index.js:137]: Ah, it seems to be a diff issue matching different lines and the old one is still below?

File: index.js [hunk 2]
```diff
@@ -293,9 +284,12 @@ This will become a hard error in the future.`,
         ms.prepend(opts.loader);
       }
 
+      const newCode = ms.toString();
+      const hasCodeChanged = code !== newCode;
```
Comment (RReverser) [inline index.js:288]: By this time we're guaranteed to modify the code, at least via unconditional `ms.remove` and `ms.prepend` just before this check, no?

Comment (davidclark87) [reply index.js:288]: @RReverser not necessarily - the unconditional remove/prepend only modify the code if you're using a custom amd function name. Majority of our project's files fell into the `hasCodeChanged === false` path which is how this made such a difference to our bundle time when using this plugin.

Comment (RReverser) [reply index.js:288]: Hm, ok. Either way, as mentioned above - can you please split those changes off into a separate PR? Them being in one PR & commit makes it harder to reason about them.

Comment (davidclark87) [reply index.js:288]: Unfortunately I'm no longer using the plugin - would you like to take over these changes? FWIW, the source map change only touches 5 lines at the end of renderChunk, if that helps clarify

Comment (RReverser) [reply index.js:288]: Ok, I'll look into that. Thanks for the PR.

File: index.js [hunk 3]
```diff
@@ -251,23 +242,23 @@ This will become a hard error in the future.`,
       }
       const ms = new MagicString(code);
 
-      tippex.match(code, workerRegexpForOutput, (fullMatch, optionsWithCommaStr, optionsStr) => {
-        let options;
```
Comment (RReverser) [inline index.js:255]: Looks like you accidentally removed this btw, so code now overrides global `options` instead.

**Comment by RReverser** [comment]
Made a minor fix, merging as-is for now but we'll definitely want to explore more robust solutions again (regex isn't one).

**Pull request merged**